src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/controller/DeliverySessionController.kt

```diff
@@ -118,4 +133,88 @@ class DeliverySessionController(
     val user = userMapper.fromToken(authentication)
     return DeliverySessionDTO.from(deliverySessionService.submitSessionFeedback(actionPlanId, sessionNumber, user))
   }
+
+  @GetMapping("/referral/{referralId}/delivery-session-appointments/{appointmentId}")
+  fun getDeliverySessionAppointment(
+    @PathVariable(name = "referralId") referralId: UUID,
+    @PathVariable(name = "appointmentId") appointmentId: UUID,
+    authentication: JwtAuthenticationToken,
+  ): DeliverySessionAppointmentDTO {
+    val user = userMapper.fromToken(authentication)
+    val referral = referralService.getSentReferral(referralId) ?: throw ResponseStatusException(
+      HttpStatus.BAD_REQUEST, "sent referral not found [referralId=$referralId]"
+    )
+    referralAccessChecker.forUser(referral, user)
+    val matchingAppointment = deliverySessionService.getSessions(referralId)
+      .flatMap { session -> session.appointments.map { appointment -> Pair(session.sessionNumber, appointment) } }
+      .firstOrNull { appointment -> appointment.second.id == appointmentId }
+      ?: throw EntityNotFoundException("Delivery session appointment not found [referralId=$referralId, appointmentId=$appointmentId]")
+    return DeliverySessionAppointmentDTO.from(matchingAppointment.first, matchingAppointment.second)
+  }
+
+  @GetMapping("/referral/{referralId}/delivery-session-appointments")
+  fun getDeliverySessionAppointments(
+    @PathVariable(name = "referralId") referralId: UUID,
+    authentication: JwtAuthenticationToken,
+  ): List<DeliverySessionAppointmentDTO> {
+    val user = userMapper.fromToken(authentication)
+    val referral = referralService.getSentReferral(referralId) ?: throw ResponseStatusException(
+      HttpStatus.BAD_REQUEST, "sent referral not found [referralId=$referralId]"
+    )
+    referralAccessChecker.forUser(referral, user)
+    return deliverySessionService.getSessions(referralId)
+      .flatMap { session -> session.appointments.map { appointment -> DeliverySessionAppointmentDTO.from(session.sessionNumber, appointment) } }
```

Comment:
`deliverySessionService.getSessionAppointments(referralId)` sounds like something that wouldn't force `deliverySessionService` users to transform on their own -- is this a smell that our entity/data model isn't fit for purpose?